[Jonathan Chong Lyen Chyn] iP

JonChong98 opened the pull request:
# :diamond_shape_with_a_dot_inside: Aegis Assistant

> Remember to not forget 

The Aegis Assistant keeps track of your tasks and helps you categorize them into a next and organized list. With the help of Aegis, you can:
- Create different types of tasks
- Specify completion dates for your tasks
- Track completion of your tasks


Setting up Aegis Assistant is _**easy**_ Just follow these steps:
1. Download the jar file from [here](https://github.com/JonChong98/ip/releases/tag/v1.0)
2. Open a terminal
3. Run the command `java -jar aegis.jar`


Here's what you can expect in future updates for Aegis Assistant:
- [X] Command Line Interface update
- [ ] Graphical User Interface update
- [ ] Reminders update

___

By default, the name of the file created for storing your tasks locally is called "aegis.txt". If you wish to change the name or location of the file, you can edit the file and directory path in Aegis.java, inside the initialSetup() function:

```java
public class Aegis {
...
    private static void initialSetup() {
        ...
        storage = new Storage("<preferred directory path>", 
                "<preferred file path>");
        ...
    }
...
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

JonChong98 committed:
Duke.java: Add basic greeting with new chatbot name

JonChong98 committed:
Duke.java: Add command echoing and exit command

JonChong98 committed:
Duke.java: Add ability to store and return text input as numbered list

JonChong98 committed:
Duke.java: Add ability to mark and unmark task completion with new Task class

JonChong98 committed:
Add ToDo, Event, Deadline task types with tracking support

JonChong98 committed:
Add DukeException.java

JonChong98 committed:
Update input.txt and EXPECTED.txt with new test case

JonChong98 committed:
Duke.java: Add error handling when trying to execute commands without required parameters

JonChong98 committed:
Duke.java: Add ability to delete tasks from the list

JonChong98 committed:
Duke.java: Add initialSetup() to populate an initial HashSet with all valid commands

JonChong98 committed:
Duke.java: Move command execution code to new executeCommand()

main() was overloaded so executeCommand() was created to house the
main command execution code in its own function. This reduces bloat
in main() and makes the code more readable.

JonChong98 committed:
Duke.java: Move program exit code to new exitProgram()

JonChong98 committed:
Duke.java: Refactor executeCommand() to unify StringTokenizer calls for command arguments

JonChong98 committed:
Duke.java: Move code for printing task list to new printTaskList()

JonChong98 committed:
Duke.java: Move code for marking tasks to new markTask()

JonChong98 committed:
Duke.java: Move code for unmarking tasks to unmarkTask()

JonChong98 committed:
Duke.java: Move code for creating todo task to new createToDoTask()

JonChong98 committed:
Duke.java: Move code for creating deadline task to new createDeadlineTask()

JonChong98 committed:
Duke.java: Move code for creating event task to new createEventTask()

JonChong98 committed:
Duke.java: Move code for deleting task to new deleteTask()

JonChong98 committed:
Duke.java: Delete leftover code in executeCommand() for marking task

JonChong98 committed:
Duke.java: Refactor executeCommand() to use switch case

JonChong98 committed:
Duke.java: Add java.io imports for file writing and reading

JonChong98 committed:
Duke.java: Add writeTaskListData() function

JonChong98 committed:
Task.java: Add toTaskSaveString()

toString() in Task.java produced a string that was hard to parse
back into a task object. A new function was needed to produce a
string that could be easily parsed to create a Task object.

JonChong98 committed:
Task.java: Add getStatusInt()

Moved the conversion for the isDone field into an int into its own
getStatusInt() function so that subclasses can call it for their own
overridden toSaveString() method and not repeat code.

JonChong98 committed:
ToDo.java: Add overridden toTaskSaveString()

JonChong98 committed:
Deadline.java: Add overridden toTaskSaveString()

JonChong98 committed:
Event.java: Add overridden toTaskSaveString()

JonChong98 committed:
Duke.java: Edit writeTaskListData() to use toTaskSaveString()

JonChong98 committed:
Duke.java: Add write tasks to file functionality in executeCommand()

JonChong98 committed:
Duke.java: Add functions to read and reconstruct tasks from data.txt

JonChong98 committed:
Duke.java: Change wildcard import for java.io to individual  imports

IDE auto changed the individual imports added a few commits back into a
wildcard imports in the previous commit unknowingly. This changes it
back to individual imports to adhere to Java coding standards.

JonChong98 committed:
Duke.java: Add directory check and creation to readTaskListData()

JonChong98 committed:
.gitignore: Add src/main/data folder

JonChong98 committed:
Add java.time imports to Duke.java, Event.java, Deadline.java

JonChong98 committed:
Deadline.java: Change by field to use LocalDate

JonChong98 committed:
Deadline.java: Add functions to generate date strings

2 different functions were created to generate different date strings.
This was done to fit the 2 different needs:
1. String to display in the list during progam runtime
2. String for saving the date in a parse-able format in data.txt

JonChong98 committed:
Event.java: Change start and end fields to use LocalDate

JonChong98 committed:
Event.java: Add functions to generate date strings

JonChong98 committed:
Add Ui, Storage, Parser, TaskList classes

hiivan commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,306 @@
+                    + "| | | | | | | |/ / _ \\\n"
+                    + "| |_| | |_| |   <  __/\n"
+                    + "|____/ \\__,_|_|\\_\\___|\n";
+        System.out.println(logo);
```
Should this be abstracted out to another class for better OOP e.g. Ui class?

hiivan commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,30 @@
+    @Override
+    public String toString() {
+        return "[D]" + super.toString() + " (by: " + this.generateByDateString() + ")";
+    }
```
I like how you overrided the toString method to print out the appropriate message rather than having another method to print that message.

hiivan commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,306 @@
+            arguments += st.nextToken() + " ";
+        }
+        arguments = arguments.trim();
+        switch (identifier) {
```
I like the use of the `enum` data structure here!

hiivan commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,306 @@
+                "delete"));
+        try {
+            readTaskListData();
+        } catch (FileNotFoundException e) {
```
Robust error handling. Good job!

hiivan reviewed (commented):
Generally employs great coding practices. Can consider refactoring the code to extract out closely related code as classes.

gingerbreaf commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,34 @@
+public class Task {
```
Would it be better if this class is abstract?

gingerbreaf commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,306 @@
+import java.util.StringTokenizer;
+
 public class Duke {
+    private static String gap = "____________________________________________________________\n";
```
This variable can be made final / constant to avoid possible confusion

gingerbreaf commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,306 @@
+        System.out.println("Total task count: " + tasks.size() + ".\n" + gap);
+    }
+
+    private static void exitProgram() {
```
Good method naming so far

gingerbreaf commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,306 @@
+                + selectedTask.toString() + "\n" + gap);
+    }
+
+    private static void createToDoTask(String arguments) throws DukeException {
```
I think it is better to separate these operations into respective files/ classes. It feels very confusing when there are so many methods in one file

gingerbreaf commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,306 @@
+    }
+
+    private static void readTaskListData() throws FileNotFoundException, IOException {
+        String filePath = "./src/main/data/duke.txt";
+        String directoryPath = "./src/main/data";
```
Good usage of camelCase

shawnnlimm commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,306 @@
+
+    private static void initialSetup() {
+        tasks = new ArrayList<Task>();
+        validCommands = new HashSet<String>();
+        validCommands.addAll(Arrays.asList("bye",
+                "list",
+                "mark",
+                "unmark",
+                "todo",
+                "deadline",
+                "event",
+                "delete"));
```
Would it be better to implement commands as an enum instead?

shawnnlimm commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,306 @@
+        }
+    }
+
+    private static void createEventTask(String arguments) throws DukeException {
```
Good use of camelCase and method naming. It is clear and easy to understand.

shawnnlimm reviewed (commented):
Overall, there are no major coding standard issues in your code. Perhaps adding some documentation to your methods would be nice. Good job!

JonChong98 committed:
Move input and ouput functions from Duke to Ui class

JonChong98 committed:
Move task list manipulation functions from Duke to TaskList class

JonChong98 committed:
Duke.java: Move Ui object initialization to initialSetup()

JonChong98 committed:
TaskList.java: Add line break to printTaskList()

JonChong98 committed:
Move file read and write functions from Duke to Storage class

JonChong98 committed:
Storage.java: Remove FileNotFoundException import

JonChong98 committed:
Duke.java: Remove redundant imports

JonChong98 committed:
Move command parsing functions from Duke to Parser class

JonChong98 committed:
Move all classes into package duke

JonChong98 committed:
Merge branch 'add-gradle-support'
Merge branch add-gradle-support into branch master

JonChong98 committed:
Duke.java: Add error catching for invalid ArrayList index

JonChong98 committed:
Add TaskListTest.java to test markTask()

JonChong98 committed:
Duke.java: Add error catching to event and deadline create functions

Morninggloryyy reviewed (commented):
LGTM! I haven't found any code standard related issue so far. Good job! :))

JonChong98 committed:
.gitignore: Add src/test/data directory

JonChong98 committed:
Add StorageTest.java to test WriteTaskListData()

JonChong98 committed:
build.gradle: Add mainClassName property for jar build

JonChong98 committed:
Deadline.java: Add JavaDoc comments

JonChong98 committed:
Deadline.java: Add missing punctuation to certain JavaDoc comments

JonChong98 committed:
Deadline.java: Add JavaDoc comment for toString()

JonChong98 committed:
Duke.java: Add JavaDoc comments

JonChong98 committed:
DukeException.java: Add JavaDoc comments

JonChong98 committed:
Event.java: Add JavaDoc comments

JonChong98 committed:
Parser.java: Add JavaDoc comments

JonChong98 committed:
Storage.java: Add JavaDoc comments

JonChong98 committed:
TaskList.java: Add JavaDoc comments

JonChong98 committed:
ToDo.java: Add JavaDoc comments

JonChong98 committed:
Ui.java: Add JavaDoc comments

JonChong98 committed:
Duke.java: Reformat code to fit Java coding standards

JonChong98 committed:
Parser.java: Reformat code to fit Java coding standards

JonChong98 committed:
Parser.java: Remove blank lines at end of file

JonChong98 committed:
Storage.java: Reformat code to fit Java coding standards

JonChong98 committed:
Reformat Task and TaskList classes to fit Java coding standards

TaskList called a method in Task which had its name changed to fit
Java coding standards, so both had to be changed at once to preserve
functionality.

JonChong98 committed:
Parser.java: Add find as a valid command

JonChong98 committed:
Task.java: Add checkDescription()

JonChong98 committed:
TaskList.java: Add printTasksWithKeyword()

JonChong98 committed:
Duke.java: Add printIdentifiedTasks()

JonChong98 committed:
Duke.java: Move printTaskCount() calls into individual methods

Moving the calls into individual methods allows for more specific
control compared to when it was a general call in executeCommand().

JonChong98 committed:
Merge branch 'branch-A-CodingStandard'

JonChong98 committed:
Merge branch 'branch-Level-9'

JonChong98 committed:
Duke.java: Remove leftover merge conflict code

JonChong98 committed:
Change all program name references in project to Aegis

JonChong98 committed:
Add dependencies and Launcher class for JavaFX GUI

JonChong98 committed:
Add basic JavaFX GUI chat window

JonChong98 committed:
Add text echoing and profile images to chat window

JonChong98 committed:
Add FXML and Java classes to handle JavaFX GUI

JonChong98 committed:
Add CSS styling to JavaFX GUI

JonChong98 committed:
Aegis.java: Remove redundant start() function

JonChong98 committed:
Add Aegis command execution functionality to JavaFX GUI

JonChong98 committed:
build.gradle: Edit settings to build jar file for GUI

JonChong98 committed:
Add assertions for verifying variable values in certain methods

JonChong98 committed:
Add missing JavaDoc comments and refined existing comments

Task, StorageTest and TaskListTest classes were missing JavaDoc
comments for their methods.

Some comments also had outdated descriptions so they were updated.

Comments for source attribution to JavaFX tutorial were added to
files that implemented the JavaFX GUI.

JonChong98 committed:
Remove redundant package imports

There were package imports in several classes that were imported
at the start of the project but are no longer used due to how
different the project is now. These imports have been removed.

JonChong98 committed:
Merge pull request #3 from JonChong98/branch-A-CodeQuality

Add missing comments and remove redundant imports

JonChong98 committed:
Merge branch 'master' into branch-A-Assertions
Merge master branch into branch-A-Assertions

JonChong98 committed:
Merge pull request #2 from JonChong98/branch-A-Assertions

Add assertions for verifying variable values in certain methods

JonChong98 committed:
Add ability to tag tasks

JonChong98 committed:
TaskList.java: Add function to check validity of task index

JonChong98 committed:
Add new parsing function for parsing Tag command arguments

JonChong98 committed:
Edit functions for tagging task to account for invalid command

JonChong98 committed:
Add task tags when task string is generated

JonChong98 committed:
Add saving and reading of tags from file

JonChong98 committed:
Add format checks for tag string input

JonChong98 committed:
Main.java: Add Aegis name to GUI title bar

JonChong98 committed:
Add user guide using README.md

JonChong98 committed:
Aegis.java: Add constructor to create instance with custom IO path

JonChong98 committed:
Add AegisTest class to test tagTask() in Aegis class

JonChong98 committed:
TaskListTest.java: Add comment to highlight class tests Aegis CLI

JonChong98 committed:
README.md: Add section for getting started and shutting down

JonChong98 committed:
README.md: Update download link for aegis.jar file
